src/kernel/arch.zig

```diff
@@ -3,7 +3,7 @@ const builtin = @import("builtin");
 const is_test = builtin.is_test;
 const build_options = @import("build_options");
 
-pub const internals = if (is_test) @import("../../test/mock/kernel/arch_mock.zig") else switch (builtin.cpu.arch) {
+pub const internals = if (is_test) @import("arch_mock") else switch (builtin.cpu.arch) {
     .i386 => @import("arch/x86/arch.zig"),
```

Comment:
If you're talking about line 7, no it can't because that would make pluto_pkg dependent on arch_pkg which I don't think can work. I'm trying to think of a way to remove this file entirely and just have the kernel do @import("arch") everywhere. I'm not even sure if it will be possbie to completely seperate the kernel from arch without some sort of glue in between like a common package that has no depencancies

Reply:
I really don't know though. I've never done kernel dev nor have I done much with zig, I just started working on this today haha, Thank you for taking the time to review this though

Reply:
So what I'm thinking is to make a package arch-common that both pluto and arch depend on. It would be filled with optional function pointers that return types. Then if you implement an arch, you just have to implement everything in arch-common to get that arch working under pluto. The main obvious downside to this is that pluto would need to null check anything it tries to use from arch-common. Its a lot more work that I thought it would be. I'll try that anyway, see how far I get.